### PR TITLE
Kc user checkout

### PR DESCRIPTION
**Pull request by kathleen-crowe**
## Description
Completes User Story #26 

The order show page needs some more formatting and styling, but all tests are passing and the story is finished.

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which maintains existing functionality)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Notes
```
User Story 26, Registered users can check out

As a registered user
When I add items to my cart
And I visit my cart
I see a button or link indicating that I can check out
And I click the button or link to check out and fill out order info and click create order
An order is created in the system, which has a status of "pending"
That order is associated with my user
I am taken to my orders page ("/profile/orders")
I see a flash message telling me my order was created
I see my new order listed on my profile orders page
My cart is now empty
```

## RSpec Results
```
Finished in 10.09 seconds (files took 4.34 seconds to load)
152 examples, 0 failures

Coverage report generated for RSpec to /Users/kathleencarroll/turing/2module/projects/monster_shop_1911/coverage. 2331 / 233
2 LOC (99.96%) covered.
```

### CODE REVIEW

File: app/controllers/profile/orders_controller.rb [hunk 1]
```diff
@@ -6,4 +6,33 @@ def index
...
+    if order.save
+      cart.items.each do |item,quantity|
+        order.item_orders.create({
+          item: item,
+          quantity: quantity,
+          price: item.price
+          })
+      end
+      session.delete(:cart)
+      flash[:notice] = "Your order was created."
```
Comment (d-atkins) [inline app/controllers/profile/orders_controller.rb]: (to add style)
```suggestion
      flash[:success] = "Your order was created."
```

File: app/controllers/profile/orders_controller.rb [hunk 2]
```diff
@@ -6,4 +6,33 @@ def index
...
+      redirect_to "/profile/orders"
+    else
+      flash[:notice] = "Please complete address form to create an order."
```
Comment (d-atkins) [inline app/controllers/profile/orders_controller.rb]:
```suggestion
      flash[:error] = "Please complete address form to create an order."
```

**Review by d-atkins — APPROVED** [review]

File: app/views/cart/show.html.erb [hunk 3]
```diff
@@ -51,7 +51,7 @@
 </table>
   <p>Total: <%= number_to_currency(cart.total) %></p>
   <p><%= link_to "Empty Cart", "/cart", method: :delete %></p>
-  <p><%= link_to "Checkout", "/orders/new", method: :get if current_user%></p>
+  <p><%= link_to "Checkout", "profile/orders/new", method: :get if current_user%></p>
```
Comment (iEv0lv3) [inline app/views/cart/show.html.erb:54]: Nicesu!

File: app/views/orders/new.html.erb [hunk 4]
```diff
@@ -22,7 +22,7 @@
 <h3 align = "right">Total: <%= number_to_currency(cart.total)%></h3>
 <center>
   <h2>Shipping Info:</h2>
-  <%= form_tag "/orders", method: :create do %>
+  <%= form_tag "/profile/orders", method: :create do %>
```
Comment (iEv0lv3) [inline app/views/orders/new.html.erb:25]: Not urgent, but creating a partial for this using `form_with` would be nice for refactoring.

File: config/routes.rb [hunk 5]
```diff
@@ -19,9 +19,7 @@
     get '/edit/pw', to: 'users#edit_pw'
     patch '/user', to: 'users#update'
     patch '/user/pw', to: 'security#update'
-    get '/orders', to: 'orders#index'
-    get '/orders/:id', to: 'orders#show'
-    patch '/orders/:id', to: 'orders#patch'
+    resources :orders, only: [:index, :show, :new, :create]
```
Comment (iEv0lv3) [inline config/routes.rb:22]: Awesome use of resources!

File: spec/factories/item.rb [hunk 6]
```diff
@@ -2,7 +2,7 @@
   factory :random_item, class: Item do
     sequence(:name)   { Faker::Commerce.product_name }
     description       { Faker::Hipster.sentence(word_count: 5) }
-    price             { Faker::Commerce.price }
+    price             { rand(0.01..100.99).to_s } #Faker::Commerce.price price above 0
```
Comment (iEv0lv3) [inline spec/factories/item.rb]: I think you can use something like this:

{ Faker::Commerce.price(range = 0.1..100.0, as_string = false) }

File: spec/features/orders/creation_spec.rb [hunk 7]
```diff
@@ -48,7 +48,12 @@
 
       new_order = Order.last
 
-      expect(current_path).to eq("/orders/#{new_order.id}")
+      # ItemOrder.find_by(item_id: @paper.id)
+      # item_order = ItemOrder.find_by(order_id: new_order.id)
```
Comment (iEv0lv3) [inline spec/features/orders/creation_spec.rb]: Is this commented text for a different test, and thus necessary?

File: spec/features/orders/creation_spec.rb [hunk 8]
```diff
@@ -82,13 +87,13 @@
         expect(page).to have_content("$2")
       end
 
-      within "#grandtotal" do
-        expect(page).to have_content("Total: $142")
-      end
+      # within "#grandtotal" do
+        expect(page).to have_content("Grand total: $142")
+      # end
```
Comment (iEv0lv3) [inline spec/features/orders/creation_spec.rb]: Is this commented text for a different test, and thus necessary?

File: spec/features/orders/creation_spec.rb [hunk 9]
```diff
@@ -82,13 +87,13 @@
...
 
-      within "#datecreated" do
-        expect(page).to have_content(new_order.created_at)
-      end
+      # within "#datecreated" do
+        expect(page).to have_content(new_order.created_at.to_formatted_s(:long))
+      # end
```
Comment (iEv0lv3) [inline spec/features/orders/creation_spec.rb]: Is this commented text for a different test, and thus necessary?

**Comment by iEv0lv3** [comment]
Nice work and thanks for being on top of the changes! There are no merge conflicts, so I'm going to merge this now.

**Pull request merged**